### DISCUSSION TITLE
InlineImage reduced quality after setting width + height options

### DISCUSSION THREAD

**Discussion by Nevinia210** [pnp/pnpcore#952] [Q&A] [answered]
Hello Everyone,
once again I'm struggling with inline images and its quality after insertion to page
Repro steps:
1. upload image to library (for example site assets) that is bigger than 400px (586px in my case)
2. create page and programmatically insert this inline image there with witdh+height set in options (width -> 586px)
3. save/publish page
4. page has blurred image that is referencing src to blob

```C#
var inlineImage = newArticle.GetInlineImage(control, HttpUtility.UrlDecode(transformedUrl), new PageImageOptions() { Alignment = PageImageAlignment.Center, Width = dimensions.Item1, Height = dimensions.Item2 });
```
when i insert inline image to page without these properties, picture is focused, but always with max 400px width
in UI it is possible to add that image there and change dimensions accordingly, picture is focused correctly than

some ideas/suggestions?

**Comment by Nevinia210** [comment, marked as answer]
Hello,
any ideas on that?

**Reply by jansenbe** [reply]
@Nevinia210 : can you do the following:

- create an as simple as possible page using the UI: one line of text with one embedded image 
- get the respective list item from the site pages library and get the content of the "CanvasContent1" field
- now do the same but create the page using code

There must be a difference in how the inline image is described in the page, you'll be able to figure that out when comparing the two versions. You'll to URL decode the retrieved page content to make it more readable (e.g. using https://www.url-encode-decode.com/).

**Reply by Nevinia210** [reply]
I restarted my pc and cleaned up all cache, and i was not able to reproduce this error, seems like something was maybe stuck in cache or something :/
anyway thanks for your advices @jansenbe
